# AI Job Market Analysis: Automation vs. Traditional Roles

## Dataset Overview

**Dataset:** AI Job Market 2026: Automation vs Traditional Role

**Source:** [Kaggle – AI Job Market 2026](https://www.kaggle.com/datasets/mariaaqdas/ai-job-market-2026-automation-vs-traditional-role)

**Description:**  
This dataset contains job postings from the United States, United Kingdom, and India. It includes information such as job titles, companies, locations, job categories, employment types, salary ranges, and whether a position is AI-related or non-AI. The dataset is used in this project to explore differences between AI-related jobs and traditional roles.

## Project Objective

The objective of this project is to analyze how artificial intelligence is influencing the job market and identify differences between AI-related and traditional roles.

Using Python and pandas, I explore salary ranges, job categories, locations, employment types, and AI-related classifications. The analysis focuses on questions such as whether AI-related positions have different salary ranges than non-AI positions, which locations offer the most AI-related opportunities, and which types of positions appear most frequently in the AI job market.

This project demonstrates practical experience with data exploration, data cleaning, filtering, aggregation, and interpreting real-world employment data.

In [7]:
import pandas as pd

df = pd.read_csv("ai_job_market_dataset.csv")

df.head()

,search_keyword,search_country,id,title,company,location,category,contract_type,contract_time,salary_min,salary_max,salary_is_predicted,created,description,is_ai_related,salary_avg,created_date,currency,country_name
0,artificial intelligence,gb,5822242641,Senior Technologist Artificial Intelligence R ...,Hackajob Ltd,"Great Baddow, Chelmsford",Engineering Jobs,permanent,NaN,40000.00,40000.00,0,2026-07-31 11:09:48+00:00,hackajob is partnering directly with BAE Syste...,True,40000.00,2026-07-31,GBP,United Kingdom
1,artificial intelligence,gb,5727313687,Artificial Intelligence,Adria Solutions,"Altrincham, Greater Manchester",IT Jobs,NaN,NaN,53702.83,53702.83,1,2026-05-12 22:05:22+00:00,AI Engineer - Manchester - Circa £50K Our clie...,True,53702.83,2026-05-12,GBP,United Kingdom
2,artificial intelligence,gb,5736675451,Artificial Intelligence Engineer,Oxford Dynamics,UK,IT Jobs,permanent,full_time,74422.25,74422.25,1,2026-05-21 21:07:04+00:00,Artificial Intelligence Engineers Contract: Pe...,True,74422.25,2026-05-21,GBP,United Kingdom
3,artificial intelligence,gb,5824444000,Artificial Intelligence Engineer,Quantiphi,"London, UK",IT Jobs,NaN,full_time,53031.54,53031.54,1,2026-08-01 18:29:25+00:00,"While technology is the heart of our business,...",True,53031.54,2026-08-01,GBP,United Kingdom
4,artificial intelligence,gb,5844156514,Artificial Intelligence Engineer,Discover International,"Monument, Central London",IT Jobs,contract,full_time,52000.00,60000.00,0,2026-08-16 11:39:19+00:00,"Salary: £52,000 - 60,000 per year Requirements...",True,56000.00,2026-08-16,GBP,United Kingdom


In [8]:
df.shape

(1696, 19)

In [9]:
df.columns

Index(['search_keyword', 'search_country', 'id', 'title', 'company',
       'location', 'category', 'contract_type', 'contract_time', 'salary_min',
       'salary_max', 'salary_is_predicted', 'created', 'description',
       'is_ai_related', 'salary_avg', 'created_date', 'currency',
       'country_name'],
      dtype='str')

In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1696 entries, 0 to 1695
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   search_keyword       1696 non-null   str    
 1   search_country       1696 non-null   str    
 2   id                   1696 non-null   int64  
 3   title                1696 non-null   str    
 4   company              1695 non-null   str    
 5   location             1696 non-null   str    
 6   category             1696 non-null   str    
 7   contract_type        432 non-null    str    
 8   contract_time        826 non-null    str    
 9   salary_min           1513 non-null   float64
 10  salary_max           1513 non-null   float64
 11  salary_is_predicted  1696 non-null   int64  
 12  created              1696 non-null   str    
 13  description          1696 non-null   str    
 14  is_ai_related        1696 non-null   bool   
 15  salary_avg           1513 non-null   float64
 16 

## 3. Dataset Description

The dataset contains 1,696 observations and 19 features. Each observation represents a job posting. Some of the key attributes include the job title, company, location, job category, contract type, minimum and maximum salary, job description, country, and whether the job is AI-related.

The dataset also contains information about the date the job was posted, the currency used for the salary, and whether the listed salary was predicted. These attributes will allow me to compare different types of jobs and examine patterns involving AI-related positions, salaries, and locations.

## 4. Dataset Format

The dataset is stored in CSV (Comma-Separated Values) format. In a CSV file, the data is organized into rows and columns, which makes it easy to load and analyze using Python libraries such as pandas.

## 5. Ease of Working With the Format

I think the CSV format makes this dataset fairly easy to work with because the information is already organized into rows and columns. Python's pandas library can also read CSV files directly, which makes it easy to view and analyze the data.

One challenge is that some values in the dataset are missing and appear as NaN in pandas. This means I may need to check for missing data before performing certain analyses, especially when working with salary or employment information.

## 6. Dataset Source and Collection

The dataset was published on Kaggle by Maria Aqdas. The job posting data was collected through the Adzuna Jobs API and includes job postings from the United States, United Kingdom, and India.

The dataset contains job postings from 2026. The data was gathered using job-search keywords, including terms related to artificial intelligence and automation. Because the information comes from online job postings available through the Adzuna API, the dataset represents jobs that were available through that source during the collection period rather than every job available in these countries.

## 7. Possible Biases

There are some possible biases in this dataset. One is that the job postings only come from the United States, United Kingdom, and India, so the dataset does not represent the entire global job market. Certain countries and types of jobs may therefore be overrepresented or underrepresented.

Another possible bias comes from the way the jobs were collected. Since the data came through the Adzuna Jobs API and specific search keywords were used, jobs that were not available through Adzuna or did not match those keywords may have been left out. I also noticed that some job postings are missing information such as salary or contract type. Because of these limitations, I would be careful about using this dataset to make conclusions about the entire job market.

## 8. Precise and Testable Questions

1. Do AI-related job postings have a higher average minimum salary than non-AI job postings in this dataset? I would compare the `salary_min` values between jobs classified as AI-related and jobs classified as non-AI.

2. Which country in the dataset has the highest number of AI-related job postings? I would count the number of AI-related postings for each country and compare the totals.

## 9. Reflection

One challenge I ran into while working with this dataset was making sure I had downloaded and opened the correct file. At first, I loaded a CSV file that had the correct column names but did not contain any usable job data. After looking more closely at the files on Kaggle, I found the full dataset and was able to load it into Jupyter Notebook using pandas. I also had to install pandas and make sure Python was looking in the correct location on my computer. These problems showed me that collecting data involves more than just downloading a file. You also have to make sure the file contains the information you expect before beginning the analysis.

Another thing I learned was the importance of inspecting a dataset before trying to answer questions with it. Using commands such as `df.head()`, `df.shape`, `df.columns`, and `df.info()` helped me understand what was actually in the dataset. I also noticed that some information was missing from certain job postings, which could affect the results of an analysis. Overall, this assignment helped me understand why checking the source, structure, and quality of the data is an important part of the data science process.